### PR TITLE
Add body contains matcher

### PR DESCRIPTION
**Pull request by exul**
Matcher to test if the body contains a substring.

### CODE REVIEW

File: src/matchers.rs [hunk 1]
```diff
@@ -432,6 +433,63 @@ impl Match for BodyExactMatcher {
...
+    BodyContainsMatcher::string(body)
+}
+
+impl Match for BodyContainsMatcher {
+    fn matches(&self, request: &Request) -> bool {
+        let body = str::from_utf8(&request.body)
+            .expect("can't convert body from byte slice to string")
+            .to_string();
+        let part =
+            str::from_utf8(&self.0).expect("can't convert expected part from byte slice to string");
```
Comment (LukeMathWalker) [inline src/matchers.rs:488]: We don't want to have `expect`/`unwrap` here. Panicking in a matcher will tear down the entire MockActor.
If any of these operations fails, we should just return false.

File: src/matchers.rs [hunk 2]
```diff
@@ -432,6 +433,63 @@ impl Match for BodyExactMatcher {
...
+///         .respond_with(ResponseTemplate::new(200))
+///         .mount(&mock_server)
+///         .await;
+///
+///     // Act
+///     let status = surf::post(&mock_server.uri())
+///         .body_string("this is a hello world example!".into())
+///         .await
+///         .unwrap()
+///         .status();
```
Comment (LukeMathWalker) [inline src/matchers.rs:459]: Probably worth adding an example with a JSON payload as well

File: src/matchers.rs (hunk 1 above)
Comment (LukeMathWalker) [reply src/matchers.rs:488]: If you want to provide good diagnostics for debugging, a `log::debug / log::trace` statement would be ideal.

File: src/matchers.rs (hunk 2 above)
Comment (exul) [reply src/matchers.rs:460]: I'm not sure if it's worth adding a JSON example, this implementation just checks for a substring, the format doesn't really matter.

**Comment by LukeMathWalker** [comment]
In it goes, thank you! :grin:

**Pull request merged**